Source/SourceKittenFramework/String+SourceKitten.swift

```diff
@@ -98,11 +104,17 @@ extension NSString {
     */
     public func NSRangeToByteRange(start start: Int, length: Int) -> NSRange? {
         let string = self as String
-        return string.byteOffsetAtIndex(start).flatMap { stringStart in
-            return string.byteOffsetAtIndex(start + length).map { stringEnd in
-                return NSRange(location: stringStart, length: stringEnd - stringStart)
-            }
+        let startUTF16Index = string.utf16.startIndex.advancedBy(start)
+        let endUTF16Index = startUTF16Index.advancedBy(length)
+        
+        guard let startUTF8Index = startUTF16Index.samePositionIn(string.utf8),
```

Comment:
Maybe caching exact results will not be efficient for most use case (I don't know other than SwiftLint) because inputs are matching results and those are almost unique.

But caching latest index might be efficient for SwiftLint.
I will test that.